ovc/resource_ovc_machine.go

```diff
@@ -349,11 +374,58 @@ func resourceOvcMachineUpdate(d *schema.ResourceData, m interface{}) error {
 		} else {
 			// delete all interfaces
 			log.Println("[DEBUG] Detaching from all external networks")
-			if err := client.Machines.DeleteExternalIP(machineIDInt, 0, ""); err != nil {
+			var emptyIP string
+			if err := client.Machines.DeleteExternalIP(machineIDInt, 0, emptyIP); err != nil {
 				return err
 			}
 		}
 	}
+	if d.HasChange("act_as_default_gateway") {
+		// get machine info in order to get cloudspace ID
+		machineInfo, err := client.Machines.Get(machineConfig.MachineID)
+		if err != nil {
+			return err
+		}
+		actAsDefaultGateway := d.Get("act_as_default_gateway").(bool)
+		if actAsDefaultGateway {
```

Comment:
Can be changed to `if d.Get("act_as_default_gateway").(bool)` and not use `actAsDefaultGateway`
But this is fine for me too